controllers/cms.py

```diff
@@ -610,8 +637,9 @@ def create_onaccept(form):
                            (T("Details"), "body"),
                            ]
             lappend = list_fields.append
+            levels = current.gis.get_relevant_hierarchy_levels()
```

Comment:
Here I modified only for bug fixing and all other modifications are done by Michael